lib/puppet/property.rb

```diff
@@ -402,6 +402,7 @@ def property_matches?(current, desired)
   # @param value [Object] the value to format as a string
   # @return [String] a pretty printing string
   def is_to_s(value)
+    value = value[0] if value.is_a?(Array) && value.size == 1
```

Comment:
We also lose the single quoted empty string for the `current` value, e.g.  `groups changed  to 'ntp'`. To repro:

{noformat}
[root@m0iobnwqvmkcte6 ~]# puppet apply -e 'user {"foobar": ensure=>present}'
Notice: Compiled catalog for m0iobnwqvmkcte6.delivery.puppetlabs.net in environment production in 0.02 seconds
Notice: /Stage[main]/Main/User[foobar]/ensure: created
Notice: Applied catalog in 0.10 seconds
[root@m0iobnwqvmkcte6 ~]# puppet apply -e 'user {"foobar": ensure=>present, groups=>['wheel']}'
Notice: Compiled catalog for m0iobnwqvmkcte6.delivery.puppetlabs.net in environment production in 0.02 seconds
Notice: /Stage[main]/Main/User[foobar]/groups: groups changed  to 'wheel'
Notice: Applied catalog in 0.11 seconds
{noformat}